unit_test_framework.h

```diff
@@ -215,9 +215,9 @@ std::ostream& print(std::ostream& os, const T& t) {
 // ----------------------------------------------------------------------------
 
 template <typename First, typename Second>
-void assert_equal(First first, Second second, int line_number);
+void assert_equal(First &&first, Second &&second, int line_number);
```

Comment:
For consistency with the rest of the file, *, &, and && should hug the type rather than the variable. The clang format settings here should take care of that if you don't want to do it manually.

Reply:
Done.